driver/src/main/scala/api/cursor.scala

```diff
@@ -202,55 +260,94 @@ object DefaultCursor {
       private def hasNext(response: Response): Boolean = response.reply.cursorID != 0
 
       @inline
-      private def hasNext(response: Response, maxDocs: Int): Boolean = {
-        response.reply.cursorID != 0 && (response.reply.numberReturned + response.reply.startingFrom) < maxDocs
-      }
+      private def hasNext(response: Response, maxDocs: Int): Boolean =
+        hasNext(response) && (response.reply.numberReturned + response.reply.startingFrom) < maxDocs
 
       @inline
       private def makeIterator(response: Response) = ReplyDocumentIterator(pack)(response.reply, response.documents)
 
-      @inline
-      private def makeRequest(implicit ctx: ExecutionContext): Future[Response] =
-        Failover2(mongoConnection, failoverStrategy) { () =>
-          mongoConnection.sendExpectingResponse(RequestMaker(query, documents, readPreference), isMongo26WriteOp)
-        }.future
+    @inline
+    private def makeRequest(implicit ctx: ExecutionContext): Future[Response] =
+      Failover2(mongoConnection, failoverStrategy) { () =>
+        mongoConnection.sendExpectingResponse(RequestMaker(query, documents, readPreference), isMongo26WriteOp)
+      }.future
+
+    @inline
+    private def isTailable =
+      (query.flags & QueryFlags.TailableCursor) == QueryFlags.TailableCursor
+
+    /** Returns next response using tailable mode */
+    private def tailResponse(current: Response, maxDocs: Int)(implicit context: ExecutionContext): Future[Option[Response]] = mongoConnection.killed flatMap {
+      case true =>
+        logger.warn("[tailResponse] Connection is killed")
+        Future.successful(Option.empty[Response])
```

Comment:
For now, doesn't fail if connection is closed there, *just* raise that there is no more response.